extension/src/fileSystem/index.ts

```diff
@@ -200,14 +200,60 @@ stages:
   return appendFileSync(dvcYamlPath, pipeline)
 }
 
+const loadYamlAsDoc = (
+  path: string
+): { doc: yaml.Document; lineCounter: yaml.LineCounter } | undefined => {
+  try {
+    const lineCounter = new yaml.LineCounter()
+    return {
+      doc: yaml.parseDocument(readFileSync(path, 'utf8'), { lineCounter }),
+      lineCounter
+    }
+  } catch {
+    Logger.error(`failed to load yaml ${path}`)
+  }
+}
+
+const getYamlFileIndent = (lines: string[]) => {
+  const firstLineWithInd = lines.find(line => line.startsWith(' ')) || ''
+  const spacesMatches = firstLineWithInd.match(/^( +)[^ ]/)
+  const spaces = spacesMatches?.[1]
+
```

Comment:
This appeared to be the simplest way to get the indent length.

Reply:
Potentially different sections could have different indentations. Can we match the plots section as a priority (if it exists)

Reply:
Updated the code to search the plots section for indent spaces if plots exists!